esphome/__main__.py

```diff
@@ -650,6 +670,13 @@ def parse_args(argv):
         "configuration", help="Your YAML configuration file directories.", nargs="+"
     )
 
+    parser_idedata = subparsers.add_parser(
+        "idedata", help="Fetch or generate the idedata and spit it out."
+    )
```

Comment:
```suggestion
    parser_idedata = subparsers.add_parser("idedata")
```

This way it won't show up in `esphome --help` (I don't think we need to/should expose this to the user, like `esphome vscode` command too)

Reply:
Ah yeah